bundles/org.eclipse.equinox.p2.publisher.eclipse/src/org/eclipse/equinox/p2/publisher/eclipse/ProductAction.java

```diff
@@ -263,4 +275,52 @@ protected File getExecutablesLocation() {
 			return new File(source);
 		return null;
 	}
+
+	private class RootFeatureAdvice implements IFilterAdvice {
```

Comment:
It could only be reused in a subclass of ProductAction, in which case protected is sufficient I think?  Though it highlights all the private methods that also cannot be specialized nor reused...